File: ios/NHSOnline/NHSOnline/Data/RootService.swift
import Foundation

struct RootService: KnownService, Codable {
    var url: String
    var javaScriptInteractionMode: JavaScriptInteractionMode
    var menuTab: MenuTab
    var integrationLevel: IntegrationLevel
    var validateSession: Bool
    var subServices: [SubService]?

    init(url: String,
         javaScriptInteractionMode: JavaScriptInteractionMode,
         menuTab: MenuTab,
         integrationLevel: IntegrationLevel,
         validateSession: Bool,
         subServices: [SubService]?) {
        self.url = url
        self.javaScriptInteractionMode = javaScriptInteractionMode
        self.menuTab = menuTab
        self.integrationLevel = integrationLevel
        self.validateSession = validateSession
        self.subServices = subServices
    }
}

File: android/app/src/main/java/com/nhs/online/nhsonline/services/knownservices/RootService.kt
package com.nhs.online.nhsonline.services.knownservices

import com.nhs.online.nhsonline.services.knownservices.enums.JavaScriptInteractionMode
import com.nhs.online.nhsonline.services.knownservices.enums.IntegrationLevel
import com.nhs.online.nhsonline.services.knownservices.enums.MenuTab
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class RootService(
        override var requiresAssertedLoginIdentity: Boolean,
        override var validateSession: Boolean,
        override var menuTab: MenuTab,
        override var javaScriptInteractionMode: JavaScriptInteractionMode,
        override var showSpinner: Boolean,
        override var integrationLevel: IntegrationLevel,
        var url: String,
        var subServices: List<SubService>? = null
): KnownService
